### DISCUSSION TITLE
Repositories are not working as expected

### DISCUSSION THREAD

**Discussion by HauXun** [community/community#19672] [Projects and Issues] [answered]
I have some problems with my repositories, my laptop just crashed hard drive and lost data, then I tried to restore them but it seems my archive is no longer working as usual, even some text files are encoded as binary characters, is there any solution to get the repositories working properly again and what is the problem here!

![Problems with repositories](https://user-images.githubusercontent.com/78190363/176381366-93e9726d-f293-41ec-a118-d2d5b3dd3bcf.png)

**Comment by thomscoder** [comment, marked as answer]
Hi @HauXun
have you tried to re-clone from the remote? This would make you lose all the unpushed/modified stuff you've been working on.
<hr>
Although, first I'd try (to try not losing the unpushed/modified commits): <br>

```
$ find .git/objects/ -size 0 -exec rm -f {} \;
$ git fetch origin
```

**Reply by HauXun** [reply]
Yes i have re-clone my repositories so i could continue my work even though all the modified files i was working on lost the commit.
Looks like your solution will work but what is the problem with my repositories here

**Reply by amangit1314** [reply]
Hi @HauXun have you tried to re-clone from the remote? I think that would help you loase all the unpushed/modified stuff you've beeen working on

**Reply by HauXun** [reply]
Yes, I solved it.
So do you know what is the problem with my repository here?

**Reply by thomscoder** [reply]
> Yes i have re-clone my repositories so i could continue my work even though all the modified files i was working on lost the commit. Looks like your solution will work but what is the problem with my repositories here

I don't know what caused your problem but keep in mind that git `.git/objects` are where git objects are stored.
Git objects are:
- Commits
- Trees
- Blob
- Tags
All these objects are linked together through a clever hash system. For example a commit object contains the tree object hash, parent commit hash, author, committer, date and message.

You can see them by running `git cat-file -p <HASH>` or `git cat-file -t <HASH>` to see the object type of that hash.

Tree object contains the directory listing for a particular commit. Tree directory contains
this directory listing gives us the hash of the Blob object associated to file snapshot or another tree for subdirectories.
etc....
As you can see they all linked and altering even one of those hash makes the whole building fall apart. This is as far as my knowledge goes, but if you want to learn more about git internals check [curios git](https://matthew-brett.github.io/curious-git/curious_git.html)

**Reply by HauXun** [reply]
Thank you for your help, I will find more information about the [curious git](https://matthew-brett.github.io/curious-git/curious_git.html) you suggest to understand more 💖💖💖
